q2_moshpit/prodigal/prodigal.py

```diff
@@ -0,0 +1,73 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2023, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+import re
+import os
+import copy as cp
+from .._utils import run_command
+from q2_types_genomics.feature_data import MAGSequencesDirFmt
+from q2_types_genomics.genome_data import (
+    LociDirectoryFormat, GenesDirectoryFormat, ProteinsDirectoryFormat,
+)
+
+
+def predict_genes_prodigal(
+        mags: MAGSequencesDirFmt,
+        translation_table_number: int = 11,
+) -> (LociDirectoryFormat, GenesDirectoryFormat, ProteinsDirectoryFormat):
+
+    # Instantiate output directories
+    loci = LociDirectoryFormat()
+    genes = GenesDirectoryFormat()
+    proteins = ProteinsDirectoryFormat()
+
+    # Pattern of fasta files in input dir
+    pattern = re.compile(
+        r"^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-4[0-9a-fA-F]{3}-"
+        r"[89abAB][0-9a-fA-F]{3}-[0-9a-fA-F]{12}\.(fa|fasta)$"
+    )
+
+    # Look for fasta files in input dir
+    fasta_files = [
+        file for file in os.listdir(mags.path) if pattern.match(file)
+    ]
```

Comment:
There is no need for the pattern matching here: the semantic type is taking care of that so when you receive the `MAGSequencesDirFmt` it is guaranteed it will only have the right FASTA files so you can directly use them.

Reply:
I see. then I'll just look for files that end with fa or fasta to collect the paths. There is an exception clause right after these lines that goes off when no files are found. Should I remove that too seeing that this is something that qiime should also control for?

Reply:
Actually, that's a good point - that should not be needed, you're right 👌.